lockfile_generator/src/yarn.rs

```diff
@@ -21,4 +22,30 @@ impl Generator for Yarn {
     fn tool(&self) -> &'static str {
         "Yarn"
     }
+
+    fn check_prerequisites(&self, manifest_path: &Path) -> Result<()> {
+        let yarn_version = yarn_version(manifest_path)?;
+        if yarn_version.starts_with("1.") {
+            let version = yarn_version.trim().into();
+            return Err(Error::UnsupportedCommandVersion("yarn", "2.0.0+", version));
```

Comment:
The error message here becomes something like:

```
unsupported yarn version 1.22.19, expected 2.0.0+
```

We are leaving it up to the user to figure out how to upgrade. Which isn't trivial because `yarn` is complicated...

But also, I don't understand `yarn` well enough to know what to suggest here... So I guess this will have to do.

Reply:
I feel like "how to update yarn from v1 to v2" is something that should be trivial to find online with extremely detailed information on how to go about it.

So I don't think it's going to be worth it to try and provide our own documentation here.